js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+                if (Precise.stringLt (currencyWithdrawMax, withdrawMax)) {
+                    currencyWithdrawMax = withdrawMax;
+                }
+                if (isDepositEnabled) {
+                    depositEnabled = true;
+                }
+                if (isWithdrawEnabled) {
+                    withdrawEnabled = true;
+                }
+                networks[network] = {
+                    'info': chain,
+                    'id': networkId,
+                    'network': network,
+                    'active': active,
+                    'deposit': isDepositEnabled,
+                    'withdraw': isWithdrawEnabled,
+                    'fee': this.safeNumber (chain, 'fee'),
+                    'precision': precision,
+                    'limits': {
+                        'withdraw': {
+                            'min': withdrawMin,
+                            'max': withdrawMax,
+                        },
+                    },
+                };
+            }
+            const networkKeys = Object.keys (networks);
+            const networkKeysLength = networkKeys.length;
+            if ((networkKeysLength === 1) || ('NONE' in networks)) {
+                const defaultNetwork = this.safeValue2 (networks, 'NONE', networkKeysLength - 1);
+                if (defaultNetwork !== undefined) {
+                    currencyFee = defaultNetwork['fee'];
+                    currencyPrecision = defaultNetwork['precision'];
+                }
+            }
+            result[code] = {
+                'id': id,
+                'code': code,
+                'info': currency,
+                'name': name,
+                'active': currencyActive,
+                'deposit': depositEnabled,
+                'withdraw': withdrawEnabled,
+                'fee': currencyFee,
+                'precision': currencyPrecision,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': currencyWithdrawMin,
+                        'max': currencyWithdrawMax,
+                    },
+                },
+                'networks': networks,
+            };
+        }
+        return result;
+    }
+
+    safeNetwork (networkId) {
+        if (networkId.indexOf ('BSC') >= 0) {
+            return 'BEP20';
+        }
+        const parts = networkId.split (' ');
+        networkId = parts.join ('');
+        networkId = networkId.replace ('-20', '20');
+        const networksById = {
+            'ETH': 'ETH',
+            'ERC20': 'ERC20',
+            'BEP20(BSC)': 'BEP20',
+            'TRX': 'TRC20',
+        };
+        return this.safeString (networksById, networkId, networkId);
+    }
+
+    async fetchMarkets (params = {}) {
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchMarkets', undefined, params);
```

Comment:
@ttodua Seems to me that we could distinguish spot and swap markets from the exchange response as marketIds and the response structure are different.

Do you see foresee any trouble if we decided to load swap + spot at the same time?

Reply:
Thanks for notes, I'll start reviewing them today.
about markets combination - in a month or so, they are going to release new Futures Api, so at this moment, I would refrain from merging them, as the structure might need to be changed again in a month or so.  I thought we could wait till that and then decide after new API is integrated.

Reply:
@ttodua let's prioritize this PR after solving Kraken if possible. 

> about markets combination - in a month or so, they are going to release new Futures Api, so at this moment, I would refrain from merging them, as the structure might need to be changed again in a month or so. I thought we could wait till that and then decide after new API is integrated.

I suggest that we implement the loading of all types at once for now and later rewrite it when the exchange adds a new version of the API. It could take a month or longer before that happens, so we should not wait in this case.